src/Streamly/Internal/LZ4.hs

```diff
@@ -296,6 +296,7 @@ compressD i0 (Stream.Stream step0 state0) =
         r <- step0 gst st
         case r of
             Stream.Yield arr st1 -> do
+                assertM (Array.byteLength arr < 2 * 1024 * 1024 * 1024)
```

Comment:
assert won't work in production code. We need to use `error` so that instead of silent buggy behavior we error out.

Reply:
I don't know if this is even required. I only have this for the time being. I need to check why `lz4-conduit` has this condition.